manifests/npm.pp

```diff
@@ -43,21 +45,34 @@
     default   => 'grep',
   }
 
-  $install_check = $::osfamily ? {
-    'Windows' => "${npm_path} ls --long --parseable | ${grep_command} \"${target}\\node_modules\\${install_check_package_string}\"",
-    default   => "${npm_path} ls --long --parseable | ${grep_command} \"${target}/node_modules/${install_check_package_string}\"",
+  $dirsep = $::osfamily ? {
+    'Windows' => "\\",
+    default   => '/'
   }
 
+  $list_command = "${npm_path} ls --long --parseable"
+  $install_check = "${list_command} | ${grep_command} \"${target}${dirsep}node_modules${dirsep}${install_check_package_string}\""
```

Comment:
I have mixed feelings about a using a variable for a directory separator in a string - I personally prefer explicit versus implicit values for it.

Reply:
IMHO this makes the code more readable - the string just differ by the dir seps...

Maybe let's replace it with a call to `regsubst` on Windows to replace `/` with `\\`?

Reply:
> Maybe let's replace it with a call to regsubst on Windows to replace / with \\?

That's an idea. Not sure if there will be unintended side-effects to doing this though.

Reply:
Maybe this is not even necessary...

According to https://docs.puppet.com/puppet/5.2/lang_windows_file_paths.html#when-to-use-each-kind-of-slash, we can just use slashes.

I don't have a Windows Puppet at hands to prove this true or false - anything available on your side?

Reply:
@poikilotherm Er I'm not sure if I'm much help - I haven't tested Puppet on Windows with anything for...quite some some time. Looking at the docs in more detail though, it seems as though we _do_ care about what type of slashes, because we're using cmd.exe to run the grep-alternative and cmd.exe requires backslashes.

Reply:
`/` should just work, both in puppet and in node.